compiler/noirc_evaluator/src/ssa/ir/dfg.rs

```diff
@@ -307,13 +307,20 @@ impl DataFlowGraph {
         instruction_id: InstructionId,
         ctrl_typevars: Option<Vec<Type>>,
     ) {
-        self.results.insert(instruction_id, Default::default());
-
-        // Get all of the types that this instruction produces
-        // and append them as results.
-        for typ in self.instruction_result_types(instruction_id, ctrl_typevars) {
-            self.append_result(instruction_id, typ);
-        }
+        let results = self
+            .instruction_result_types(instruction_id, ctrl_typevars)
+            .into_iter()
+            .enumerate()
+            .map(|(position, typ)| {
+                self.values.insert(Value::Instruction {
+                    typ,
+                    position,
+                    instruction: instruction_id,
+                })
+            })
+            .collect();
+
+        self.results.insert(instruction_id, results);
```

Comment:
Also includes this change which inlines an unnecessary function and avoid repeatedly pushing onto a `Vec` in favour of iterators.